impl/src/main/java/com/okta/sdk/impl/resource/AbstractPropertyRetriever.java

```diff
@@ -202,6 +202,10 @@ protected Date getDateProperty(DateProperty key) {
             return null;
         }
 
+        if(value instanceof Date) {
```

Comment:
Why is this needed now?

Reply:
This method not working properly because of

`dateFormatter.parse(String.valueOf(value));`

If `value` is a string formatted as `"yyyy-MM-dd'T'HH:mm:ss.SSSZ"` then its works fine, but when the `value` is `Date` the result of `String.valueOf(value)` is a string formatted as '"EEE MMM dd HH:mm:ss zzz yyyy"'

So this method will crash in this case